### PR TITLE
Remove "C" linkages

### PR DESCRIPTION
**Pull request by layog**
@kpet This PR is to remove all the "C" linkages as discussed in #776

### CODE REVIEW

File: test_conformance/api/test_kernel_arg_changes.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@
 //
 #include "testBase.h"
 
-extern "C" { extern cl_uint gRandomSeed;}
+extern cl_uint gRandomSeed;
```
Comment (kpet) [inline test_conformance/api/test_kernel_arg_changes.cpp:18]: `gRandomSeed` is declared in `testHarness.h`. All other declarations should be removed.

File: test_conformance/spir/datagen.cpp [hunk 2]
```diff
@@ -93,38 +93,6 @@ void DataGenerator::setArgGenerator(const KernelArgInfo& argInfo,
     m_argGenerators[argInfo.getTypeName()] = pGen;
 }
 
-float get_random_float(float low, float high, MTdata d)
```
Comment (kpet) [inline test_conformance/spir/datagen.cpp:96]: Nice :).

File: test_conformance/api/test_kernel_arg_changes.cpp (hunk 1 above)
Comment (layog) [reply test_conformance/api/test_kernel_arg_changes.cpp]: Done. I have removed the declarations from other places as well (where `extern "C"` wasn't used). Let me know if this violates some comitting standards.

Comment (kpet) [reply test_conformance/api/test_kernel_arg_changes.cpp]: Nope, makes perfect sense. Thanks for doing this!

**Pull request merged**